runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -296,11 +313,135 @@ public long getBackpointerCount() {
+     * @param queue    queue to insert discovered addresses in the given range
+     * @param startAddress range start address (inclusive)
+     * @param stopAddress  range end address (exclusive and lower than start address)
+     * @param filter       function to filter entries
+     * @param checkpoint   boolean indicating if it is a checkpoint stream
+     * @param maxGlobal    maximum Address until which we want to sync (is not necessarily equal to start address)
+     * @return
+     */
+    protected boolean getStreamAddressMap(final UUID streamId,
+                                          final NavigableSet<Long> queue,
+                                          final long startAddress,
+                                          final long stopAddress,
+                                          final Function<ILogData, BackpointerOp> filter,
+                                          final boolean checkpoint,
+                                          final long maxGlobal) {
+        // Sanity check: startAddress must be a valid address and greater than stopAddress.
+        // The startAddress refers to the max resolution we want to resolve this stream to,
+        // while stopAddress refers to the lower boundary locally resolved.
+        // If startAddress is equal to stopAddress there is nothing to resolve.
+        if (Address.isAddress(startAddress) && (startAddress > stopAddress)) {
+
+            StreamAddressSpace streamAddressSpace = null;
+
+            // If start and stop addresses are consecutive (e.g. 2 and 1) there is no need to search for this stream's
+            // address map, as there are no other addresses within this range.
+            if (startAddress - stopAddress > 1) {
+                log.trace("getStreamAddressMap[{}] from {} to {}", this, startAddress, stopAddress);
+                // This step is an optimization.
+                // Attempt to read the last entry and verify its backpointer,
+                // if this address is already resolved locally do not request the stream map to the sequencer as new
+                // updates are not required to be synced.
+                ILogData d;
+                try {
+                    log.trace("getStreamAddressMap: readAddress[{}]", startAddress);
+                    d = read(startAddress);
+                } catch (TrimmedException e) {
+                    if (options.ignoreTrimmed) {
+                        log.warn("getStreamAddressMap: Ignoring trimmed exception for address[{}]," +
+                                " stream[{}]", startAddress, streamId);
+                        return false;
+                    } else {
+                        throw e;
+                    }
+                }
+
+                boolean fetchStreamAddressMap = true;
+
+                // Verify if backpointer is an address locally resolved
+                if (!runtime.getParameters().isBackpointersDisabled() && d.hasBackpointer(streamId)) {
```

Comment:
But isn't it redundant, considering that we are checking if it has a backpointer to this stream??

Reply:
Actually in the case of regression I think backpointer is not the ground truth, it could point to some other stream. I can show you an example, let's discuss it later today to see if it's a valid case.